### PR TITLE
Nitish

### PR DESCRIPTION
**Pull request by Nitish0007**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,11 +1,19 @@
 import React from "react";
```
Comment (aashu0148) [inline src/App.js:1]: remove all the warnings from console. For firebase to auto deploy we need a clean console without warnings and errors.

File: src/App.js [hunk 2]
```diff
@@ -1,11 +1,19 @@
...
-
+import { BrowserRouter as Router, Link, Route, Routes } from "react-router-dom";
 import "styles/main.scss";
```
Comment (aashu0148) [inline src/App.js]: we are following a proper way of importing like first comes imports form libraries then comes our component imports then comes any utility imports then any asset imports and at last css imports

File: src/components/InputControl/InputControl.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  const [visible, setVisible] = useState(props.password ? false : true);
+  return (
+    <div className={styles.container}>
+      {props.label && <label className={styles.label}>{props.label}</label>}
+      <div className={styles.inputContainer}>
+        <input
+          className="basic-input"
+          type={visible ? "text" : "password"}
+          {...props}
+          ref={ref}
```
Comment (aashu0148) [inline src/components/InputControl/InputControl.js:18]: did you checked if forward ref is working with this component or not ?

File: src/pages/Merchant/LoginPage/LoginPage.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
+import React, { useRef, useState } from "react";
+
+import styles from "./LoginPage.module.scss";
```
Comment (aashu0148) [inline src/pages/Merchant/LoginPage/LoginPage.js]: same import structure is different as above

File: src/pages/Merchant/LoginPage/LoginPage.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+      ) {
+        dummyErrors.email = "Enter valid email";
+        setErrors(dummyErrors);
+      }
+    }
+    if (passRef.current.value === "") {
+      dummyErrors.password = "Enter password";
+      setErrors(dummyErrors);
+    }
+  };
```
Comment (aashu0148) [inline src/pages/Merchant/LoginPage/LoginPage.js]: here in this function why updating errors on every if check ?

we can update the errors in the last and will result in the same impact but with less state re renders.

File: src/pages/Merchant/LoginPage/LoginPage.js [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+  const handleFormSubmit = (event) => {
+    event.preventDefault();
+    validateForm();
```
Comment (aashu0148) [inline src/pages/Merchant/LoginPage/LoginPage.js]: validateFrom() will only update the errors but do not interrupt the execution of this function. take a case where email is invalid ,then this function will show error but also do not stop the further execution of handleFormSubmit function

File: src/pages/Merchant/LoginPage/LoginPage.js [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+  };
+
+  return (
+    <div className={styles.login}>
+      <div className={styles.loginLeft}>
+        <h1>Buy Best!</h1>
+        <span>100+ products available at best price</span>
+        <img
+          className={styles.loginLeftImage}
+          src="https://image.shutterstock.com/z/stock-vector-set-of-fruit-and-vegetables-logo-for-groceries-agriculture-stores-packaging-and-advertising-318421853.jpg"
```
Comment (aashu0148) [inline src/pages/Merchant/LoginPage/LoginPage.js]: never user direct url for src in img tags instead make a constant above with a proper name and then use it.

File: src/pages/Merchant/LoginPage/LoginPage.js [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+        ></img>
+        <div className={styles.loginLeftPanelIcons}></div>
```
Comment (aashu0148) [inline src/pages/Merchant/LoginPage/LoginPage.js]: why this div is there when do not have any children ?

File: src/pages/Merchant/LoginPage/LoginPage.js [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+      </div>
+      <div className={styles.loginRightPanel}>
+        <form className={styles.loginForm} onSubmit={handleFormSubmit}>
+          <div className={styles["loginRightPanel-mainBody"]}>
+            <h2>Login</h2>
+
+            <div className={styles["loginRightPanel-inputWrapper"]}>
```
Comment (aashu0148) [inline src/pages/Merchant/LoginPage/LoginPage.js]: is there a need of inputWrapper div when we already applied the required styling in the InputControl component ?

File: src/pages/Merchant/LoginPage/LoginPage.js [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+                label="Password"
+                placeholder="Enter password"
+                ref={passRef}
+                error={errors?.password}
+                password="true"
+              />
+            </div>
+            <span
+              className={styles["loginRightPanel_helper-text"]}
+              // onClick={changeURl}
```
Comment (aashu0148) [inline src/pages/Merchant/LoginPage/LoginPage.js]: why this code is commented out ?

File: src/pages/Merchant/LoginPage/LoginPage.module.scss [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+.loginRightPanel {
+  width: 50%;
+  padding: 4rem 1.5rem 0rem 1.5rem;
+  display: flex;
+  align-items: center;
+}
+.loginRightPanel-mainBody {
+  border: 1px solid $bg-light-gray;
+  height: 100%;
+  // width: 100%;
```
Comment (aashu0148) [inline src/pages/Merchant/LoginPage/LoginPage.module.scss]: why commented out this? if dont need then remove this

File: src/pages/Merchant/LoginPage/LoginPage.module.scss [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+  &:focus {
+    outline: 1px solid $bg-secondary;
+  }
+  &:hover {
+    outline: 1px solid $bg-secondary;
+  }
+  &::placeholder {
+    color: $bg-medium-gray;
+  }
+}
```
Comment (aashu0148) [inline src/pages/Merchant/LoginPage/LoginPage.module.scss]: these styling is already present in main.scss with a class name of basic-input you can dic the class to any element where you want to apply this styling

File: src/pages/Merchant/Register/Register.js [hunk 13]
```diff
@@ -0,0 +1,180 @@
+import React, { useState, useRef } from "react";
+import { useNavigate } from "react-router-dom";
+
+import styles from "./Register.module.scss";
```
Comment (aashu0148) [inline src/pages/Merchant/Register/Register.js]: import structure

File: src/pages/Merchant/Register/Register.js [hunk 14]
```diff
@@ -0,0 +1,180 @@
...
+    if (lnameRef.current.value === "") {
+      dummyErrors.lname = "Enter last name";
+      setErrors(dummyErrors);
+    }
+
+    if (mobileRef.current.value === "") {
+      dummyErrors.mobile = "Enter mobile number";
+      setErrors(dummyErrors);
+    } else {
+      if (
```
Comment (aashu0148) [inline src/pages/Merchant/Register/Register.js]: same mistake as in login page validateForm

File: src/pages/Merchant/Register/Register.js [hunk 15]
```diff
@@ -0,0 +1,180 @@
...
+      if (confirmpassRef.current.value !== passRef.current.value) {
+        dummyErrors.confirmpass = "passwords must be same";
+        setErrors(dummyErrors);
+      }
+    }
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    validateForm();
```
Comment (aashu0148) [inline src/pages/Merchant/Register/Register.js]: same mistake as in login handleSubmit function

File: src/pages/Merchant/Register/Register.js [hunk 16]
```diff
@@ -0,0 +1,180 @@
...
+  };
+
+  const navigate = useNavigate();
```
Comment (aashu0148) [inline src/pages/Merchant/Register/Register.js]: this thing should be on top with useRef s

File: src/pages/Merchant/Register/Register.js [hunk 17]
```diff
@@ -0,0 +1,180 @@
...
+  };
+
+  return (
+    <div className={styles.register}>
+      <div className={styles.registerLeftPanel}>
+        <h1>Buy Best!</h1>
+        <span>100+ products available at best price</span>
+        <img
+          className={styles.registerLeftPanelImage}
+          src="https://image.shutterstock.com/z/stock-vector-set-of-fruit-and-vegetables-logo-for-groceries-agriculture-stores-packaging-and-advertising-318421853.jpg"
```
Comment (aashu0148) [inline src/pages/Merchant/Register/Register.js]: same src mistake as in login page

File: src/pages/Merchant/Register/Register.js [hunk 18]
```diff
@@ -0,0 +1,180 @@
...
+            <div className={styles["registerRightPanel-inputWrapper"]}>
+              <InputControl
+                label="Email"
+                placeholder="Enter email"
+                ref={emailRef}
+                error={errors?.email}
+              />
+            </div>
+            <div className={styles["registerRightPanel-inputContainer"]}>
+              <div className={styles["registerRightPanel-inputWrapper"]}>
```
Comment (aashu0148) [inline src/pages/Merchant/Register/Register.js]: same inputWrapper mistake as in login page

File: src/pages/Merchant/Register/Register.js [hunk 19]
```diff
@@ -0,0 +1,180 @@
...
+                  password="true"
+                />
+              </div>
+            </div>
+            <span
+              className={styles["registerRightPanel_helper-text"]}
+              onClick={changeURl}
+            >
+              Already have an account!
+            </span>
```
Comment (aashu0148) [inline src/pages/Merchant/Register/Register.js]: this whole thing should not be clickable instead another text named "login now" should be clickable. and also did you checked that this is working ?

File: src/pages/Merchant/Register/Register.js (hunk 14 above)
Comment (Nitish0007) [reply src/pages/Merchant/Register/Register.js]: resolved

File: src/pages/Merchant/Register/Register.js (hunk 18 above)
Comment (Nitish0007) [reply src/pages/Merchant/Register/Register.js]: done

File: src/App.js (hunk 2 above)
Comment (Nitish0007) [reply src/App.js]: resolved

File: src/pages/Merchant/Register/Register.js (hunk 13 above)
Comment (Nitish0007) [reply src/pages/Merchant/Register/Register.js]: resolved

File: src/pages/Merchant/Register/Register.js (hunk 15 above)
Comment (Nitish0007) [reply src/pages/Merchant/Register/Register.js]: done

File: src/components/InputControl/InputControl.js (hunk 3 above)
Comment (Nitish0007) [reply src/components/InputControl/InputControl.js:18]: yes

**Pull request merged**